### ISSUE TITLE
Explicit set to null for collection property handling should not be there for Nvms.Always

### ISSUE THREAD

**Issue by sjaakd** [mapstruct/mapstruct#954]
When `NullValueCheckStrategy.ALWAYS` there should not be an explicit null check. 

```java

        if ( entity.getPermissions() != null ) {

            Set<Permission> set = permissionDTOSetToPermissionSet( dto.getPermissions() );
            if ( set != null ) {
                entity.getPermissions().clear();
                entity.getPermissions().addAll( set );
            }
            else {
                entity.setPermissions( null );
            }
        }

```
MapStruct handles collections different via the `addAll`. For regular property-to-property mapping NPE checks are generated for type conversions. Otherwise the source property (also if null) is mapped to the target.

When `NullValueCheckStrategy.ALWAYS` is applicable, an NPE check is always included. In that case, the source property value is left as-is: `null` for `create` methods, to its original value for `update` methods (TODO check this).

In-line with this reasoning the source for an `update method` should be left alone, and hence`else` branch should not be generated for  `NullValueCheckStrategy.ALWAYS`.

**Comment by sjaakd** [comment]
And the same goes for source presence checkers.

**Comment by sjaakd** [comment]
Done!

**Comment by gunnarmorling** [comment]
@sjaakd, shouldn't this be assigned to the 1.1.0.Final release?

**Comment by sjaakd** [comment]
> shouldn't this be assigned to the 1.1.0.Final release?

Yep.. You're right.

**Comment by t-andryukha** [comment]
For MapStruct 1.2.0.Final it still sets collection to null after explicit null check:
`
Set<PerDmtTableItemActivity> set = tableItemActivityListToPerDmtTableItemActivitySet( article.getTableItemActivities() );
        if ( set != null ) {
            perArticle.setTableItemActivities( set );
        }
        else {
            perArticle.setTableItemActivities( null );
        }
`

**Comment by filiphr** [comment]
@t-andryukha I think that what you are looking for has been fixed with #1306. For normal mappings the setting of that will no longer happen. It would be in the release notes of the upcoming 1.3.0.Beta2

**Comment by t-andryukha** [comment]
@filiphr Exactly. After upgrading to 1.3.0.Beta2 the issue has gone. Thanks.
